### PR TITLE
feat(ion-sfu): new sdk

### PR DESCRIPTION
**Pull request by tarrencev**

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -1,191 +1,110 @@
...
-    url.searchParams.append('peer', uid);
-    const transport = new WebSocketTransport(url.toString(), config.options);
-    log.setLevel(config.loglevel !== undefined ? config.loglevel : log.levels.WARN);
+  ontrack?: (track: MediaStreamTrack, stream: RemoteStream) => void;
 
-    this.knownStreams = new Map();
-    this.uid = uid;
-    this.streams = {};
+  localStreams: LocalStream[];
+  remotes: { [name: string]: RemoteStream };
```
Comment (OrlandoCo) [inline src/client.ts:20]: Can we use Map<string,RemoteStream>?, I think it's more legible to remove and iterate elements

File: src/stream.ts [hunk 2]
```diff
@@ -146,9 +148,51 @@ export class LocalStream extends MediaStream {
...
+      this.video = 'none';
+    }
+    this.select();
+  }
+
+  unmute(kind: 'audio' | 'video') {
+    if (kind === 'audio') {
+      this.audio = true;
+    } else if (kind === 'video') {
+      this.video = 'high';
```
Comment (OrlandoCo) [inline src/stream.ts:194]: Should default to last selected layer ```this.video```

File: src/stream.ts [hunk 3]
```diff
@@ -85,8 +89,41 @@ export class LocalStream extends MediaStream {
...
+          sendEncodings: this.options.encodings
+            ? this.options.encodings
+            : [
+                {
+                  rid: 'f',
+                },
+                {
+                  rid: 'h',
+                  scaleResolutionDownBy: 2.0,
+                  maxBitrate: 150000,
```
Comment (OrlandoCo) [inline src/stream.ts:109]: we should make the maxbitrate configurable with defaults.

Comment (tarrencev) [reply src/stream.ts:109]: it is possible to define them by passing encodings on the `Constraints` object

```
export interface Constraints extends MediaStreamConstraints {
  resolution: string;
  codec: string;
  simulcast: boolean;
  encodings?: RTCRtpEncodingParameters[];
}
```

Comment (OrlandoCo) [reply src/stream.ts:109]: Yes, but Im not sure if we should expose encodings, since rids are hardcoded in sever. also encodings orders matters.

Comment (tarrencev) [reply src/stream.ts:109]: ok. let me rethink the api then

**Pull request merged**